drivers/timer/nrf_grtc_timer.c

```diff
@@ -0,0 +1,571 @@
+		.handler = NULL,
+		.p_context = NULL,
+		.channel = chan,
+	};
+	nrfx_err_t result;
+
+	IS_CHANNEL_ALLOWED_ASSERT(chan);
+
+	/* Set the CC value to mark channel as not triggered and also to enable it
+	 * (makes CCEN=1). COUNTER_SPAN is used so as not to fire an event unnecessarily
+	 * - it can be assumed that such a large value will never be reached.
+	 */
+	result = nrfx_grtc_syscounter_cc_absolute_set(&user_channel_data, COUNTER_SPAN, false);
+
+	if (result != NRFX_SUCCESS) {
+		return -EPERM;
+	}
+
+	return 0;
+}
+
+int z_nrf_grtc_timer_capture_read(int32_t chan, uint64_t *captured_time)
+{
+	/* TODO: The implementation should probably go to nrfx_grtc and this
+	 *       should be just a wrapper for some nrfx_grtc_syscounter_capture_read.
+	 */
+
+	uint64_t capt_time;
+
+	IS_CHANNEL_ALLOWED_ASSERT(chan);
+
+	/* TODO: Use `nrfy_grtc_sys_counter_enable_check` when available (NRFX-2480) */
+	if (NRF_GRTC->CC[chan].CCEN == GRTC_CC_CCEN_ACTIVE_Enable) {
+		/* If the channel is enabled (.CCEN), it means that there was no capture
+		 * triggering event.
+		 */
+		return -EBUSY;
+	}
+
+	capt_time = nrfy_grtc_sys_counter_cc_get(NRF_GRTC, chan);
+
+	__ASSERT_NO_MSG(capt_time < COUNTER_SPAN);
+
+	*captured_time = capt_time;
+
+	return 0;
+}
+
+#if defined(CONFIG_NRF_GRTC_SLEEP_ALLOWED)
+int z_nrf_grtc_wakeup_prepare(uint64_t wake_time_us)
+{
+	nrfx_err_t err_code;
+	static uint8_t systemoff_channel;
+	uint64_t now = counter();
+	/* Minimum time that ensures valid execution of system-off procedure. */
+	uint32_t minimum_latency_us =
+		nrfy_grtc_waketime_get(NRF_GRTC) + nrfy_grtc_timeout_get(NRF_GRTC) + 1000;
+	uint32_t chan;
+	int ret;
+	k_spinlock_key_t key = k_spin_lock(&lock);
+
+	err_code = nrfx_grtc_channel_alloc(&systemoff_channel);
+	if (err_code != NRFX_SUCCESS) {
+		k_spin_unlock(&lock, key);
+		return -ENOMEM;
+	}
+	(void)nrfx_grtc_syscounter_cc_int_disable(systemoff_channel);
+	ret = compare_set(systemoff_channel, now + wake_time_us, NULL, NULL);
+	if (ret < 0) {
+		k_spin_unlock(&lock, key);
+		return ret;
+	}
+
+	for (uint32_t grtc_chan_mask = NRFX_GRTC_CONFIG_ALLOWED_CC_CHANNELS_MASK;
+	     grtc_chan_mask > 0; grtc_chan_mask &= ~BIT(chan)) {
+		/* Clear all GRTC channels except the systemoff_channel. */
+		chan = u32_count_trailing_zeros(grtc_chan_mask);
+		if (chan != systemoff_channel) {
+			nrfx_grtc_syscounter_cc_disable(chan);
+		}
+	}
+
+	/* Make sure that wake_time_us was not triggered yet. */
+	if (nrfy_grtc_sys_counter_compare_event_check(NRF_GRTC, systemoff_channel)) {
+		k_spin_unlock(&lock, key);
+		return -EINVAL;
+	}
+	/* Make sure that it is not too close to wake up event trigger. */
+	else if (now + minimum_latency_us > get_comparator(systemoff_channel)) {
+		k_spin_unlock(&lock, key);
+		return -EINVAL;
+	}
+
+	/* This mechanism ensures that stored CC value is latched. */
+	uint32_t wait_time =
+		nrfy_grtc_timeout_get(NRF_GRTC) * CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / 32768 +
```

Comment:
shouldn't 32768 come from some system configuration file (dts/header)?

Reply:
CC @adamkondraciuk

Reply:
There is no such configuration